networkx/algorithms/community/community.py

```diff
@@ -0,0 +1,79 @@
+#-*- coding: utf-8 -*-
+import networkx as nx
+
+__all__ = ['girvan_newman']
+
+
+def girvan_newman(G, weight=None):
+    """Find communities in graph using Girvan-Newman method.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+
+    weight : string, optional (default=None)
+       Edge data key corresponding to the edge weight.
+
+    Returns
+    -------
+    List of tuples which contains the clusters of nodes.
+
+    Examples
+    --------
+    >>> G = nx.path_graph(10)
+    >>> comp = girvan_newman(G)
+    >>> comp[0]
+    ([0, 1, 2, 3, 4], [8, 9, 5, 6, 7])
+
+    Notes
+    -----
+    The Girvan–Newman algorithm detects communities by progressively removing
+    edges from the original graph. Algorithm removes edge with the highest
+    betweenness centrality at each step. As the graph breaks down into pieces,
+    the tightly knit community structure is exposed and result can be depicted
+    as a dendrogram.
+    """
+    g = G.copy().to_undirected()
+    components = []
+    while g.number_of_edges() > 0:
+        _remove_max_edge(g, weight)
+        components.append(_communities(g),)
+    return components
+
+
+def _remove_max_edge(G, weight=None):
```

Comment:
Maybe this would be better named with plural `edges` instead of singular `edge`, since many edges are removed.